mne_connectivity/tests/test_wsmi.py

```diff
@@ -0,0 +1,582 @@
+    conn_array_equiv = wsmi(data, kernel=3, tau=1, sfreq=sfreq, names=ch_names)
+
+    assert_allclose(conn_epochs.get_data(), conn_array_equiv.get_data(), rtol=1e-10)
+
+
+def test_wsmi_deterministic():
+    """Test that wSMI produces deterministic results."""
+    sfreq = 100.0
+    n_epochs, n_channels, n_times = 2, 3, 150
+
+    # Create identical datasets
+    data = np.zeros((n_epochs, n_channels, n_times))
+    t = np.linspace(0, n_times / sfreq, n_times)
+    for epoch in range(n_epochs):
+        for ch in range(n_channels):
+            data[epoch, ch, :] = np.sin(2 * np.pi * (8 + ch * 3) * t)
+
+    ch_names = ["A", "B", "C"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+
+    epochs1 = EpochsArray(data.copy(), info, tmin=0.0)
+    epochs2 = EpochsArray(data.copy(), info, tmin=0.0)
+
+    # Compute connectivity with identical parameters
+    conn1 = wsmi(epochs1, kernel=3, tau=1)
+    conn2 = wsmi(epochs2, kernel=3, tau=1)
+
+    # Results should be identical
+    assert_array_equal(conn1.get_data(), conn2.get_data())
+
+
+def test_wsmi_anti_aliasing():
+    """Test anti-aliasing parameter."""
+    sfreq = 100.0
+    n_epochs, n_times = 2, 200
+    data = np.zeros((n_epochs, 3, n_times))
+
+    t = np.linspace(0, n_times / sfreq, n_times)
+    for epoch in range(n_epochs):
+        data[epoch, 0, :] = np.sin(2 * np.pi * 10 * t)
+        data[epoch, 1, :] = np.sin(2 * np.pi * 12 * t)
+        data[epoch, 2, :] = np.sin(2 * np.pi * 15 * t)
+
+    ch_names = ["A", "B", "C"]
+    info = create_info(ch_names, sfreq=sfreq, ch_types="eeg")
+    epochs = EpochsArray(data, info, tmin=0.0)
+
+    # Test with anti-aliasing enabled
+    conn_with_filter = wsmi(epochs, kernel=3, tau=1, anti_aliasing=True)
+
+    # Test with anti-aliasing disabled (should produce warning)
+    import warnings
+
+    with warnings.catch_warnings(record=True) as w:
+        warnings.simplefilter("always")
+        conn_without_filter = wsmi(epochs, kernel=3, tau=1, anti_aliasing=False)
+        assert len(w) > 0
+        assert "Anti-aliasing disabled" in str(w[0].message)
+
+    # Both should produce valid results
+    assert np.all(np.isfinite(conn_with_filter.get_data()))
+    assert np.all(np.isfinite(conn_without_filter.get_data()))
+
+
+# Ground truth validation tests (simplified)
+def _load_test_data():
+    """Load the wsmi test data."""
+    import os
+    import pickle
+
+    test_data_path = os.path.join(
+        os.path.dirname(__file__), "data", "wsmi_test_data.pkl"
+    )
+
+    if not os.path.exists(test_data_path):
+        raise FileNotFoundError(
+            f"Required wSMI test data not found at {test_data_path}. "
+            "Ground truth validation tests cannot run without this data."
+        )
+
+    with open(test_data_path, "rb") as f:
+        test_data = pickle.load(f)
+
+    return test_data
+
+
+def test_wsmi_ground_truth_validation():
+    """Test wSMI against ground truth data for regression testing."""
```

Comment:
mention here where the ground truth data comes from (IIUC it's the original implementation provided with the paper?)